sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -485,6 +507,28 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     assert(e4.contains("Invalid join type in joinWith: " + LeftAnti.sql))
   }
 
+  test("joinWith left_outer") {
```

Comment:
I realized there's an existing left join test, so I'll remove this new test case and fold in the schema checks into the existing case.